src/Raven.Server/Routing/RequestRouter.cs

```diff
@@ -114,11 +110,26 @@ public async ValueTask HandlePath(RequestHandlerContext reqCtx)
                     skipAuthorization = context.Request.Method == "OPTIONS";
                 }
 
-                if (_ravenServer.Configuration.Security.AuthenticationEnabled && skipAuthorization == false)
+                var status = RavenServer.AuthenticationStatus.ClusterAdmin;
+                bool authResult = true;
+                try
                 {
-                    var authResult = TryAuthorize(tryMatch.Value, context, reqCtx.Database);
-                    if (authResult == false)
-                        return;
+                    if (_ravenServer.Configuration.Security.AuthenticationEnabled && skipAuthorization == false)
+                    {
+                        authResult = TryAuthorize(tryMatch.Value, context, reqCtx.Database, out status);
+                        if (authResult == false)
+                            return;
+                    }
+                }
+                finally
+                {
+                    if (tryMatch.Value.SkipLastRequestTimeUpdate == false)
+                    {
+                        var now = SystemTime.UtcNow;
+                        _ravenServer.Statistics.LastRequestTime = now;
+                        if (skipAuthorization == false && authResult && status != RavenServer.AuthenticationStatus.ClusterAdmin)
+                            _ravenServer.Statistics.LastAuthorizedNonClusterAdminRequestTime = now;
```

Comment:
I wonder if we shouldn't be setting these operations only once every 1 - 5 seconds.
The reason is that otherwise under load, we may create a hotspot.
Another issue is whatever in multi threaded code, we'll have tearing of the the value in some cases

Reply:
Hmm... can setting a property cause a hotpath? It's the same what we already have for any request:
`_ravenServer.Statistics.LastRequestTime = now;`

and never got problems with that.

You're right that value might be not accurate in some multi threading cases although discrepancy won't be big so I think we can afford it here

Reply:
You are writing to the same memory location from multiple threads, may cause sync costs / cache fights between cores.
We can also do the `LastRequestCheck` every 15 sec

Reply:
Fixed in https://github.com/ravendb/ravendb/pull/9717